claat/fetch.go

```diff
@@ -227,22 +228,49 @@ func fetchDriveFile(id string, nometa bool) (*resource, error) {
 
 var crcTable = crc64.MakeTable(crc64.ECMA)
 
-func slurpBytes(client *http.Client, dir, url string, n int) (string, error) {
-	res, err := retryGet(client, url, n)
+func slurpBytes(client *http.Client, codelabSrc, dir, imgURL string, n int) (string, error) {
+	// images can be local in Markdown cases or remote.
+	// Only proceed a simple copy on local reference.
+	var b []byte
+	var ext string
+	u, err := url.Parse(imgURL)
 	if err != nil {
 		return "", err
 	}
-	defer res.Body.Close()
-	b, err := ioutil.ReadAll(res.Body)
+	if u.Host == "" {
+		cdir, err := filepath.Abs(path.Dir(codelabSrc))
```

Comment:
I may sound a bit paranoid but it would be nice if you extracted this part, sanitizing local file path, into a separate function and wrote a test with all possible edge cases. I would also use `testing/quick` in addition to manual test cases.